apps/docs/src/components/Search/Markprompt.tsx

```diff
@@ -0,0 +1,126 @@
+import { searchActive } from './searchStore'
+import { Markprompt } from '@markprompt/react'
+import { h, Component } from 'preact'
+import { useEffect, useRef, useState } from 'preact/hooks'
+import hljs from 'highlight.js'
+//@ts-ignore
+import hljs_svelte from 'highlightjs-svelte'
+
+function Search({ visible }: { visible: boolean }) {
+  const container = useRef(null)
+
+  const callback = (mutationList: any) => {
+    for (const mutation of mutationList) {
+      if (
+        mutation.type === 'attributes' &&
+        mutation.attributeName === 'data-loading-state' &&
+        mutation.target.className === 'MarkpromptContentPlain'
+      ) {
+        const value = mutation.target.attributes.getNamedItem('data-loading-state')
+
+        if (value.nodeValue === 'done') {
+          mutation.target.querySelectorAll('pre').forEach((el: HTMLElement) => {
+            if (!el.classList.contains('hljs') && el.innerHTML.length > 20) {
+              hljs.highlightElement(el)
+            }
+          })
+        }
+      }
+    }
+  }
+
+  const config = { attributes: true, childList: true, subtree: true }
+
+  useEffect(() => {
+    const observer = new MutationObserver(callback)
+
+    if (container.current) {
+      observer.observe(container.current, config)
+    }
+
+    return () => observer.disconnect()
+  }, [container.current])
```

Comment:
refs aren't valid dependencies because mutating them doesn't re-render the component, so the dependency array at this point is just `[]`.